pre-analysis/COMMON_dilution.py

```diff
@@ -0,0 +1,291 @@
+
+            containers = process.output_containers()
+
+            if containers[0].type.id == container_ids["96wellplate"].id:
+                sort_name = Well(artifact.location[1]).rev_zero() + "_" + artifact.name
+            else:
+#                unsort_name = inart.name
+#                print(unsort_name)
+#                sort_name = sorted(unsort_name, key=lambda item: (''.join(filter(str.isalpha, item)), int(''.join(filter(str.isdigit, item)))))
+                sort_name = inart.name
+
+            sort_name_tuple = tuple(sort_name)
+
+            concentration[sort_name] = {
+                'Sample Name': inart.name,
+                'Clarity LimsID': inart.samples[0].id,
+                'Container': inart.location[0].name,
+                'Well': well,
+                'Concentration type': conType,
+                'Concentration (ng/ul)': conc,
+                'Target concentration (ng/ul)': targetConc,
+                'Target volume (ul)': targetVolume,
+                'Dilution factor': dilutionFactor,
+                'Sample volume (ul)': sampleVol,
+                'Buffer volume (ul)': buffVol,
+                'Analysis': analysis,
+            }
+
+
+    return concentration
+
+
+def create_lines_QIAgility(concentration):
+    """Write one line for each sampleid.
+    Add concentration for each sampleid.
+    """
+
+    logging.info("Creating lines")
+
+    lines = [SEPARATOR["Separator semicolon"].join(HEADER["Header QIAgility"])]
+
+    for sample_name, sample_info in concentration.items():
+        lines.append(f"{sample_info['Sample Name']}{SEPARATOR['Separator semicolon']}{sample_info['Concentration (ng/ul)']:.2f}")
+
+    return lines
+
+
+def create_lines_Beckman(concentration):
+    """Write one line for each well in a 96 well plate. 
+    Add concentration for wells with samples.
+    """
+
+    logging.info("Creating lines")
+
+    lines = [SEPARATOR["Separator tab"].join(HEADER["Header Beckman"])]
+
+    for sample_data, sample_info in concentration.items():
+        lines.append(f"{sample_info['Well']}{SEPARATOR['Separator tab']}{sample_info['Concentration (ng/ul)']:.2f}")
+
+    return lines
+
+
+def create_lines_Manual(concentration):
+    """ Write dilution file. 
+    Add new line for each sampleid.
+    """
+
+    logging.info("Create lines")
+
+    lines = [SEPARATOR["Separator semicolon"].join(HEADER["Header Manual"])]
+
+    for sample_data in concentration.values():
+        sorted_sample_data = [sample_data[header] for header in HEADER["Header Manual"]]
+        line = SEPARATOR["Separator semicolon"].join(str(data) for data in sorted_sample_data)
+        lines.append(line)
+
+    return lines
+
+
+def write_to_file_txt(args, lines):
+    """Write lines to file
+    """
+    filename = args["fileNAME"]+"_dilution-set-up.txt" 
+    logging.info(f"Writing to file {filename}")
+    with open(filename, "w") as dilution_file:
+        dilution_file.write("\n".join(lines))
```

Comment:
Have file_name as an input instead of args so that the file type can be changed and the name more custom.
```suggestion
def write_to_file_txt(file_name, lines):
    """Write lines to file
    """
    logging.info(f"Writing to file {filename}")
    with open(filename, "w") as dilution_file:
        dilution_file.write("\n".join(lines))
```